### PR TITLE
Initial commit for source code to build as3 Declaration

### PR DESCRIPTION
**Pull request by veeramalla-f5**
This code builds the AS3 Declaration with pool members.

  1.  I have created a new file with name buildAS3Declaration.go
  2. There are no changes(Addition and Deletion) to the existing code
  3. Unit tests will be added 
  
Known Issues:
  1. I need to handle the scenario where user provide absolute reference path of the pool instead of pool name itself.

### CODE REVIEW

File: pkg/appmanager/as3Manager.go [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+type pool []Member
+type serviceName string
+type appName string
+type tenantName string
+type tenant map[appName][]serviceName
+type as3Object map[tenantName]tenant
+
+// Fake Function to validate the buildAS3Delaration method
+func getEndpoints(tenantName, appName, serviceName) pool {
+	//ip := []poolMember{"172.16.1.1", "192.16.2.2", "1.1.1.1"}
```
Comment (ssurenr) [inline pkg/appmanager/as3Manager.go]: Let's remove this and next line.

File: pkg/appmanager/as3Manager.go [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+	//return ip
+	return []Member{
+		{"1.1.1.1", 80, ""},
+		{"2.2.2.2", 80, ""},
+		{"3.3.3.3", 80, ""},
+	}
+}
+
+// Traverses through the AS3 Json using the information passed from buildAS3Declaration
```
Comment (ssurenr) [inline pkg/appmanager/as3Manager.go]:
```suggestion
// Traverses through the AS3 JSON using the information passed from buildAS3Declaration,
```

File: pkg/appmanager/as3Manager.go [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+// Parses the AS3 Json and populates it with poolmembers
```
Comment (ssurenr) [inline pkg/appmanager/as3Manager.go]:
```suggestion
// parses the AS3 JSON and populates it with pool members
```

File: pkg/appmanager/as3Manager.go [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+func updatePoolMembers(tnt tenantName, app appName, svc serviceName, ips []string, port int32, as3Json map[string]interface{}) map[string]interface{} {
```
Comment (ssurenr) [inline pkg/appmanager/as3Manager.go]: as3Json can be renamed to a name with lesser ambiguity.

File: pkg/appmanager/as3Manager.go [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+				}
+				port := eps[0].Port
+				log.Debugf("Updating AS3 Template for tenant '%s' app '%s' service '%s', ", tnt, app, svcs[svc])
+				updatePoolMembers(tnt, app, svcs[svc], ips, port, as3Json)
+			}
+		}
+	}
+
+	declaration, err := json.Marshal(as3Json)
+	if err == nil {
```
Comment (ssurenr) [inline pkg/appmanager/as3Manager.go]:
```suggestion
	if err != nil {
```

File: pkg/appmanager/as3Manager.go [hunk 6]
```diff
@@ -0,0 +1,100 @@
...
+		//log.Errorf("[as3_log] Issue marshalling AS3 Json")
```
Comment (ssurenr) [inline pkg/appmanager/as3Manager.go]: Let's also log the `err` content.

File: pkg/appmanager/as3Manager.go [hunk 7]
```diff
@@ -0,0 +1,100 @@
...
+	}
+	log.Debugf("[as3_log] AS3 Template is populated with the pool members")
+	log.Debugf("[as3_log] Printing AS3 Template ...")
```
Comment (ssurenr) [inline pkg/appmanager/as3Manager.go]: Let's remove this line.

File: pkg/appmanager/as3Manager.go (hunk 4 above)
Comment (veeramalla-f5) [reply pkg/appmanager/as3Manager.go]: Will make this as a new commit instead of a inline

**Pull request merged**